### PR TITLE
create navigation menu that is dynamic for small and large screens

### LINKED ISSUE #20: Make navigation bar

### PR DESCRIPTION
**Pull request by zalhabash**
Create a navbar that stays at the top of the screen when you scroll. When the screen is resized below 1024px width, the pages you can navigate to are instead put in a menu that pops up when you press on the hamburger icon in the top right corner.
Closes #20

### CODE REVIEW

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -1,12 +1,28 @@
-<script>
+<script lang="ts">
 	import '$lib/styles.scss';
 	import { page } from '$app/stores';
+	import Navbar from '$lib/Navbar.svelte';
+	import logoUrl from './logo.webp';
+
+	const pages = [
+		{ name: 'Calendar', url: '/calendar' },
+		{ name: 'About', url: '/about' },
```
Comment (nstringham) [inline src/routes/+layout.svelte:9]: I noticed this page is called "about us" on the page itself but it only says "about" here. Should we make that consistent?

File: src/lib/Navbar.svelte [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+	let menuExpanded = false;
+</script>
+
+<header>
+	<div class="container">
+		<nav>
+			<ul>
+				<li>
+					<a href="/" class="brand contrast"><img src={logoUrl} alt="" />{siteName}</a>
```
Comment (nstringham) [inline src/lib/Navbar.svelte:18]: Clicking this should also close the menu

File: src/lib/Navbar.svelte [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+	@media (width < 1024px) {
+		li:has(button) {
+			padding: 0;
+		}
+
+		button {
+			appearance: none;
+			background: none;
+			border: none;
```
Comment (nstringham) [inline src/lib/Navbar.svelte:91]: Buttons have blue text by default on iOS
IMG_0397
You can fix this with `color: inherit`.

**Review by nstringham — CHANGES_REQUESTED** [review]

File: src/lib/Navbar.svelte (hunk 3 above)
Comment (zalhabash) [reply src/lib/Navbar.svelte:93]: fixed in [ddc6a1d](https://github.com/kendoclub-at-umich/website/pull/40/commits/ddc6a1de116a629025ac7d03d2b6d7732ee857df)

File: src/lib/Navbar.svelte (hunk 2 above)
Comment (zalhabash) [reply src/lib/Navbar.svelte]: fixed in [5a1c5f0](https://github.com/kendoclub-at-umich/website/pull/40/commits/5a1c5f06d6191bba80372956b4c8541e5264eb5a)

File: src/routes/+layout.svelte (hunk 1 above)
Comment (zalhabash) [reply src/routes/+layout.svelte]: Yeah we should, although part of me feels like it's weird to have "About Us" and "Contact Us" right on top of each other, like the lengths of those are so close that they look awkward next to each other, but maybe that's just me?
Still, it makes more sense than a different alternative, I think, and we're adding at least one more page anyway and possibly removing the Contact Us page, so it should be fine.
Changed page name to "About Us" in [68bc3d5](https://github.com/kendoclub-at-umich/website/pull/40/commits/68bc3d59f58cf413d92294f849b8e017564f3fbf)

File: src/lib/Navbar.svelte (hunk 2 above)
Comment (zalhabash) [reply src/lib/Navbar.svelte]: Also thanks for giving me the chance to figure out how to do that one on my own. I kind of panicked at first, but calmed myself down and realized I had everything I needed to figure this out bc I understand the code.

Comment (nstringham) [reply src/lib/Navbar.svelte]: I knew you could figure it out. You got close but your code also opens the menu when you click the brand link. It should probably only close it.

File: src/routes/+layout.svelte (hunk 1 above)
Comment (nstringham) [reply src/routes/+layout.svelte]: In general I would say content should take priority over styling but it's good that your thinking about these things (as long as you aren't overthinking 😉)

**Review by nstringham — CHANGES_REQUESTED** [review]
See respond to previous comments

File: src/lib/Navbar.svelte (hunk 3 above)
Comment (nstringham) [reply src/lib/Navbar.svelte:93]: Looks good
IMG_0398

File: src/lib/Navbar.svelte (hunk 2 above)
Comment (zalhabash) [reply src/lib/Navbar.svelte]: Oooooops 😂
Fixed in [faa773e](https://github.com/kendoclub-at-umich/website/pull/40/commits/faa773ecdb5912c82253d218f2a7559880e5695e)

**Review by nstringham — APPROVED** [review]

**Pull request merged**